swagger_server/itm/itm_scene.py

```diff
@@ -157,11 +162,75 @@ def _actions_condition_met(self, actions_condition) -> bool:
                 return True
         return False
 
+    # True if any of the specified supplies is <= the specified quantity
+    def _supply_condition_met(self, supply_conditions :List[Supplies], session_state :State) -> bool:
```

Comment:
Everything worked great for me. I just learned how to use the any() function for conditions over the weekend though so here is a suggestion that you can take or leave if you're interested and don't think it would hurt maintainability. 

```
def _supply_condition_met(self, supply_conditions: List[Supplies], session_state: State) -> bool:
        if not supply_conditions:
            return True

        supply_conditions_met = any(
            state_supply.quantity <= supply_condition.quantity 
            for supply_condition in supply_conditions 
            for state_supply in session_state.supplies 
            if supply_condition.type == state_supply.type
        )
       if supply_conditions_met:
          print(f'_supply_condition_met: a supply condition is met; returning true')
       else:
          print(f'_supply_condition_met: did not find a supply meeting criteria; returning false')

       return supply_conditions_met
```

Reply:
Love it.  I bet I have other places I can use something like that.